Userland/Libraries/LibLine/Editor.cpp

```diff
@@ -1779,7 +1779,7 @@ enum VTState {
     BracketArgsSemi = 7,
     Title = 9,
 };
-static VTState actual_rendered_string_length_step(StringMetrics& metrics, size_t index, StringMetrics::LineMetrics& current_line, u32 c, u32 next_c, VTState state, Optional<Style::Mask> const& mask);
+static VTState actual_rendered_string_length_step(StringMetrics& metrics, size_t index, StringMetrics::LineMetrics& current_line, u32 c, u32 next_c, VTState state, Optional<Style::Mask> const& mask, Optional<size_t> const& maximum_length_width = {}, Optional<size_t&> last_return = {});
```

Comment:
This is always called with all its arguments anyway, why do the new ones have defaults?

Reply:
Not in `Editor::actual_rendered_string_metrics(Utf32View ...)`.
Do you want me to add `maximum_line_width` to this overload?

Reply:
oh I missed that one, nah, it's alright.